examples/pytorch/graphsage/node_classification.py

```diff
@@ -31,29 +36,32 @@ def forward(self, blocks, x):
         return h
 
     def inference(self, g, device, batch_size, num_workers, buffer_device=None):
-        # The difference between this inference function and the one in the official
-        # example is that the intermediate results can also benefit from prefetching.
-        g.ndata['h'] = g.ndata['feat']
-        sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1, prefetch_node_feats=['h'])
+        feat = g.ndata['feat']
+        sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1, prefetch_node_feats=['feat'])
         dataloader = dgl.dataloading.NodeDataLoader(
                 g, torch.arange(g.num_nodes()).to(g.device), sampler, device=device,
-                batch_size=batch_size, shuffle=False, drop_last=False, num_workers=num_workers,
-                persistent_workers=(num_workers > 0))
+                batch_size=batch_size, shuffle=False, drop_last=False,
+                num_workers=num_workers)
+
         if buffer_device is None:
             buffer_device = device
 
         for l, layer in enumerate(self.layers):
-            y = torch.zeros(
+            y = torch.empty(
                 g.num_nodes(), self.n_hidden if l != len(self.layers) - 1 else self.n_classes,
-                device=buffer_device)
+                device=buffer_device, pin_memory=True)
```

Comment:
In the `link_pred` example, `pin_memory` is set to be `args.pure_gpu`. Why different?